### PR TITLE
Refactor Bib & footprint cells - part 1

### PR DESCRIPTION
**Pull request by reaperrr**
This PR tries to get stuff out of the way that is easy, but would blow the number of commits and changed lines out of proportion on #13553.

Adds `ITargetableCells` interface for use in #13553, renames `Bib`, and removes the bib-related hacks from `Building` and `FootprintUtils`, while adding a `LocalCenterOffset` to `Building`, to allow shifting the actor center to account for bibs (or other visuals, if modders chose to use it that way).
Finally, moves the renamed `WithBuildingBib` to Mods.Cnc.

Dependency for update of #13553.

### CODE REVIEW

**Comment by reaperrr** [comment]
Note: The reason I didn't move the bib trait to Mods.Cnc is the preview init stuff, which I didn't feel like messing with.

File: OpenRA.Game/Traits/TraitsInterfaces.cs
Comment (pchote) [inline OpenRA.Game/Traits/TraitsInterfaces.cs]: Please split this to its own interface.  `IOccupySpace` is part of our lowest-level ActorMap code, which should remain separate from targeting.

**Comment by pchote** [comment]
> Note: The reason I didn't move the bib trait to Mods.Cnc is the preview init stuff, which I didn't feel like messing with.

I fixed that for you in #13562.

**Comment by reaperrr** [comment]
Updated and rebased on #13562.

**Comment by reaperrr** [comment]
Rebased.

**Comment by reaperrr** [comment]
Before this is approved/merged, I'm not 100% sure how the 'bib' cells of the footprints were treated before.
They definitely need to be treated like '=' for placement purposes, but I'm not 100% sure about pathfinding, because if the pathfinder doesn't consider them 'pathable', densely packed bases might spell trouble for the pathfinder.

Should we maybe introduce a `-` cell type that behaves like `=` for placement/occupaction, but like `_` for pathfinding, and use that for bib cells?
I can probably move the `FootprintCellType` enum to this PR to do this, to keep temporary hacks to a minimum.

I just need approval of that idea first, before I go through the hassle of updating the upgrade rule and yaml changes.

File: OpenRA.Mods.Cnc/Traits/Render/WithBuildingBib.cs
Comment (pchote) [inline OpenRA.Mods.Cnc/Traits/Render/WithBuildingBib.cs]: Bonus points if you can remove this flag and use the x/y offsets in the sequences instead.

File: mods/cnc/rules/structures.yaml
Comment (pchote) [inline mods/cnc/rules/structures.yaml]: 420?  Why not 512 (half a cell)?

File: OpenRA.Mods.Cnc/Traits/Render/WithBuildingBib.cs
Comment (pchote) [inline OpenRA.Mods.Cnc/Traits/Render/WithBuildingBib.cs]: A simpler fix would be to remove this flag, forcing the `false` behavior for all uses of this trait, and then change the mini bib actors to use a normal idle overlay.

A future PR can then clean up the positioning and turn this into a general trait for rendering tiled artwork (e.g. TD/RA bridges, d2 buildings).

File: mods/cnc/rules/structures.yaml
Comment (reaperrr) [inline mods/cnc/rules/structures.yaml]: Because of `BodyOrientation.CameraPitch` not being 255 (same 'issue' as with vertical HitSHape.TargetableOffsets). But I just noticed it should actually have been 840 in that case.

File: OpenRA.Mods.Cnc/Traits/Render/WithBuildingBib.cs
Comment (reaperrr) [inline OpenRA.Mods.Cnc/Traits/Render/WithBuildingBib.cs]: I haven't checked yet, but don't mini-bibs use tiled artwork as well? In that case we'd have to go with the first fix, at least for mini-bibs larger than a single tile.

File: mods/cnc/rules/structures.yaml
Comment (reaperrr) [inline mods/cnc/rules/structures.yaml]: Actually you're right, these offsets don't take CameraPitch into consideration since they're not rotated like the TargetableOffsets are, so -512 is the correct value. Will fix it.

**Comment by reaperrr** [comment]
Updated.
As discussed on IRC, I've deferred the HasMinibib property removal to a follow-up, to make this PR a bit easier to review and get it merged more quickly.

File: OpenRA.Mods.Common/Traits/Buildings/Building.cs
Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/Buildings/Building.cs]: Note to reviewers: This duplication is intended for now, #13553 will update this part as soon as this PR is merged.
This is primarily a measure to keep the diff of the next revision of #13553 in check, as code-wise that will be harder to review than this PR.

File: OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs
Comment (abcdefg30) [inline OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs]: Wrong indentation here.

File: mods/d2k/rules/structures.yaml
Comment (abcdefg30) [inline mods/d2k/rules/structures.yaml]: This breaks Atreides05 as the smuggler frigate can't land anymore:
frigateblocked

**Review by abcdefg30 — COMMENTED** [review]
Looks good to me otherwise. 👍 once those comments have been resolved.

File: mods/d2k/rules/structures.yaml
Comment (reaperrr) [inline mods/d2k/rules/structures.yaml]: Oh, I see. In that case I'll drop that commit, I just didn't expect this '=' to have a purpose.

**Comment by reaperrr** [comment]
Updated.

File: OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs
Comment (pchote) [inline OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs]: This will break for mods that use >4 cell width actors.  You should be able to use something like `+ " " + string.Concat(Enumerable.Repeat("=", newDim.X))`

**Review by pchote — APPROVED** [review]
Looks good as far as I can tell aside from that one nit.  Feel free to merge once you update that.

**Comment by reaperrr** [comment]
Updated.

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/dfac297240049ade166444df8797dd1ebdb9f004...d8ec9ef2d64bceaaf300a23fcb648f662b541b6b

**Pull request merged**
